### PR TITLE
[Merged by Bors] - feat(NumberTheory/ArithmeticFunction): Miscellaneous lemmas about multiplicative arithmetic functions

### PR DESCRIPTION
**Pull request by amellendijk**
These are three small lemmas that were helpful while developing the Selberg sieve.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/NumberTheory/ArithmeticFunction.lean [hunk 1]
```diff
@@ -590,6 +590,15 @@ theorem map_prod_of_subset_primeFactors [CommSemiring R] {f : ArithmeticFunction
     f (∏ a ∈ t, a) = ∏ a ∈ t, f a :=
   map_prod_of_prime h_mult t fun _ a => prime_of_mem_primeFactors (ht a)
 
+theorem map_div_of_squarefree [CommGroupWithZero R] {f : ArithmeticFunction R}
+    (h_mult : IsMultiplicative f) {l d : ℕ} (hdl : d ∣ l) (hl : Squarefree l) (hd : f d ≠ 0) :
```
Comment (b-mehta) [inline Mathlib/NumberTheory/ArithmeticFunction.lean:594]: Might it be better to relax `Squarefree l` to `(l / d).Coprime d` here? That allows things like `f (36 / 9)`

File: Mathlib/NumberTheory/ArithmeticFunction.lean [hunk 2]
```diff
@@ -771,6 +780,19 @@ theorem lcm_apply_mul_gcd_apply [CommMonoidWithZero R] {f : ArithmeticFunction R
     apply Finset.inter_subset_union
   · simp [factorization_lcm hx hy]
 
+theorem map_lcm [CommGroupWithZero R] {f : ArithmeticFunction R}
```
Comment (b-mehta) [inline Mathlib/NumberTheory/ArithmeticFunction.lean:783]: `map_gcd`, to match?

File: Mathlib/NumberTheory/ArithmeticFunction.lean [hunk 3]
```diff
@@ -771,6 +780,19 @@ theorem lcm_apply_mul_gcd_apply [CommMonoidWithZero R] {f : ArithmeticFunction R
...
+    (h_mult : f.IsMultiplicative) {x y : ℕ} (hf : f (x.gcd y) ≠ 0) :
+    f (x.lcm y) = f x * f y / f (x.gcd y) := by
+  rw [←h_mult.lcm_apply_mul_gcd_apply, mul_div_cancel_right₀ _ hf]
+
+theorem eq_zero_of_squarefree_of_dvd_eq_zero [CommMonoidWithZero R] {f : ArithmeticFunction R}
+    (h_mult : IsMultiplicative f) {m n : ℕ} (h_sq : Squarefree n) (hmn : m ∣ n)
```
Comment (b-mehta) [inline Mathlib/NumberTheory/ArithmeticFunction.lean:789]:
```suggestion
    (hf : IsMultiplicative f) {m n : ℕ} (h_sq : Squarefree n) (hmn : m ∣ n)
```
this file usually uses `hf` for this assumption. 
(Similarly I'd personally use `hn : Squarefree n`, but that's optional)

**Review by b-mehta — APPROVED** [review]
bors d+

**Comment by FLDutchmann** [comment]
bors merge

**Pull request closed without merging**